libsupport/include/katana/ArrowVisitor.h

```diff
@@ -204,40 +205,24 @@ struct ArrowDispatcher {
     auto type_id =
         GetArrowTypeID(std::get<index>(std::forward_as_tuple(args...)));
 
-    switch (type_id) {
-#define TYPE_CASE(EnumType)                                                    \
-  case arrow::Type::EnumType: {                                                \
+#define ENUM_TYPE_CASE(z, n, unused)                                           \
+  case static_cast<arrow::Type::type>(n): {                                    \
     using ArrowType =                                                          \
-        typename arrow::TypeIdTraits<arrow::Type::EnumType>::Type;             \
+        typename arrow::TypeIdTraits<static_cast<arrow::Type::type>(n)>::Type; \
     return Call<ArrowTypes..., ArrowType>(                                     \
         std::forward<Visitor>(visitor), std::forward<Args>(args)...);          \
   }
-      TYPE_CASE(INT8)
-      TYPE_CASE(UINT8)
-      TYPE_CASE(INT16)
-      TYPE_CASE(UINT16)
-      TYPE_CASE(INT32)
-      TYPE_CASE(UINT32)
-      TYPE_CASE(INT64)
-      TYPE_CASE(UINT64)
-      TYPE_CASE(FLOAT)
-      TYPE_CASE(DOUBLE)
-      TYPE_CASE(BOOL)
-      TYPE_CASE(DATE32)     // since UNIX epoch in days
-      TYPE_CASE(DATE64)     // since UNIX epoch in millis
-      TYPE_CASE(TIME32)     // since midnight in seconds or millis
-      TYPE_CASE(TIME64)     // since midnight in micros or nanos
-      TYPE_CASE(TIMESTAMP)  // since UNIX epoch in seconds or smaller
-      TYPE_CASE(STRING)     // TODO(daniel) DEPRECATED
-      TYPE_CASE(LARGE_STRING)
-      TYPE_CASE(STRUCT)
-      TYPE_CASE(LIST)  // TODO(daniel) DEPRECATED
-      TYPE_CASE(LARGE_LIST)
-      TYPE_CASE(NA)
-#undef TYPE_CASE
+
+    switch (type_id) {
+      // 37 in 4.0
+      // 38 in 6.0
+      static_assert(arrow::Type::MAX_ID == 38);
+      BOOST_PP_REPEAT(38, ENUM_TYPE_CASE, ~);
```

Comment:
add comment that we are switching over all arrow types, as this is no longer clear